### PR TITLE
Add DecoratedDistributedTransactionAdmin

### PR DESCRIPTION
**Pull request by brfrn169**
## Description

This PR adds `DecoratedDistributedTransactionAdmin` which is an abstract class of decorator class for `DistributedTransactionAdmin`.

## Related issues and/or PRs

N/A

## Changes made

- Added `DecoratedDistributedTransactionAdmin`.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

File: core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import com.scalar.db.io.DataType;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public abstract class DecoratedDistributedTransactionAdmin implements DistributedTransactionAdmin {
+
+  private final DistributedTransactionAdmin decoratedDistributedTransactionAdmin;
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java:15]:
```suggestion
  private final DistributedTransactionAdmin distributedTransactionAdmin;
```
Should it be as above?
This class is a `DecoratedDistributedTransactionAdmin` so naming the member variable `decoratedDistributedTransactionAdmin` looks a bit weird to me.
What do you think?

**Review by feeblefakie — COMMENTED** [review]
Sorry for the late review. Looking good, but left a comment for naming. PTAL!

File: core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java (hunk 1 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java:15]: Ah, good point, thank you. Maybe the class name `DecoratedDistributedTransactionAdmin` is not good. I think `DistributedTransactionAdminDecorator` is a better name. What do you think?

Comment (brfrn169) [reply core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java]: As discussed, we will rename the field name to `distributedTransactionAdmin`. Thanks.

Comment (brfrn169) [reply core/src/main/java/com/scalar/db/common/DecoratedDistributedTransactionAdmin.java]: Fixed in 66dd2c920ae48795d471f7893b3dc79becd6513e. I've also renamed the field name in other `DecoratedXXX` classes.

@feeblefakie Could you please take a look again when you have time? Thanks.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**